src/Runner.Worker/IssueMatcher.cs

```diff
@@ -8,6 +8,28 @@ namespace GitHub.Runner.Worker
 {
     public delegate void OnMatcherChanged(object sender, MatcherChangedEventArgs e);
 
+
+    public sealed class IssueMetadata
+    {
+        public IssueMetadata(string key, string value)
+            : this(null, false, null, new []{ KeyValuePair.Create(key, value) })
+        {
+        }
+
+        public IssueMetadata(string category, bool infrastructureIssue, string logMessageOverride, IEnumerable<KeyValuePair<string, string>> data)
+        {
+            this.Category = category;
+            this.IsInfrastructureIssue = infrastructureIssue;
+            this.LogMessageOverride = logMessageOverride;
+            this.Data = data;
+        }
+
+        public readonly string Category;
+        public readonly bool IsInfrastructureIssue;
+        public readonly string LogMessageOverride;
```

Comment:
`LogMessageOverride` is transient data.  We won't persist it into any `Issue` objects we create.